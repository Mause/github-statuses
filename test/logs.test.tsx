import { render } from "@testing-library/react";
import {
  ConstructLine,
  LineWithTimestamp,
} from "~/routes/$owner/$repo/actions/$id/logs";

describe("constructLine", () => {
  it("error", () => {
    const el = render(<ConstructLine line="##[error]Error message" />);

    expect(el.container).toMatchSnapshot();
  });

  it("ansi colors", () => {
    const line = "\u001b[36;1moptions(crayon.enabled = TRUE)\u001b[0m";
    const el = render(<ConstructLine line={line} />);

    expect(el.container).toMatchSnapshot();
  });

  it("lineWithTimestamp", () => {
    let el = render(
      <LineWithTimestamp
        line={{
          line: "hello world",
          timestamp: "2022-01-01T22:00",
        }}
        showTimestamps={true}
      />,
    );
    expect(el.container).toMatchSnapshot();

    el = render(
      <LineWithTimestamp
        line={{
          line: "hello world",
          timestamp: "2022-01-01T22:00",
        }}
        showTimestamps={false}
      />,
    );
    expect(el.container).toMatchSnapshot();
  });
});
