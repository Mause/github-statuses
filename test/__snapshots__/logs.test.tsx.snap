// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`constructLine > ansi 256 colors 1`] = `
<div>
  <span
    style="color: rgb(128, 128, 128);"
  >
    test_selector:
  </span>
</div>
`;

exports[`constructLine > ansi colors 1`] = `
<div>
  <span
    style="color: rgb(0, 187, 187); font-weight: bolder;"
  >
    options(crayon.enabled = TRUE)
  </span>
</div>
`;

exports[`constructLine > error 1`] = `
<div>
  <span
    style="color: rgb(255, 83, 83);"
  >
    Error message
  </span>
</div>
`;

exports[`constructLine > lineWithTimestamp > false 1`] = `
<div>
  <span>
    hello world
  </span>
</div>
`;

exports[`constructLine > lineWithTimestamp > true 1`] = `
<div>
  <span>
    <time
      datetime="2022-01-01T22:00"
    >
      2022-01-01T22:00
    </time>
    Â 
    <span>
      hello world
    </span>
  </span>
</div>
`;
